### PR TITLE
fix: auto-discover assets after wallet onboarding completion

### PR DESCRIPTION
**Pull request by josheleonard**
Resolves [Race condition between fetching token registry and asset discovery on wallet restore](https://github.com/brave/brave-browser/issues/35696#top)

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Restore wallet using QA wallet
2. tokens should be auto discovered when first viewing the portfolio overview page

### CODE REVIEW

**Review by Douglashdaniel — APPROVED** [review]
++

File: components/brave_wallet_ui/page/screens/onboarding/onboarding-success/onboarding-success.tsx [hunk 1]
```diff
@@ -69,8 +71,11 @@ export const OnboardingSuccess = () => {
 
   // effects
   React.useEffect(() => {
+    // now that the token registry is populated, discover assets
+    discoverAssets()
+
     report(BraveWallet.OnboardingAction.Complete)
-  }, [report])
+  }, [report, discoverAssets])
```
Comment (supermassive) [inline components/brave_wallet_ui/page/screens/onboarding/onboarding-success/onboarding-success.tsx:78]: afaik this is not needed as mutations should be stable functions
https://github.com/reduxjs/redux-toolkit/blob/ed9fc19027a31afee2234dea54fb36c8beef46ad/packages/toolkit/src/query/react/buildHooks.ts#L1074-L1080

File: components/brave_wallet/browser/brave_wallet_service.cc [hunk 2]
```diff
@@ -1281,7 +1281,7 @@ void BraveWalletService::Base58Encode(
 }
 
 void BraveWalletService::DiscoverAssetsOnAllSupportedChains() {
-  DiscoverAssetsOnAllSupportedChains(false);
+  DiscoverAssetsOnAllSupportedChains(true);
```
Comment (supermassive) [inline components/brave_wallet/browser/brave_wallet_service.cc:1284]: This is also called on `WalletActions.refreshAll` and now will bypass rate_limit which is not what we might want

Comment (josheleonard) [reply components/brave_wallet/browser/brave_wallet_service.cc]: Updated. PTAL

**Review by supermassive — APPROVED** [review]
Wallet core lgtm

File: ios/brave-ios/Sources/BraveWallet/Preview Content/MockBraveWalletService.swift [hunk 3]
```diff
@@ -292,7 +292,7 @@ class MockBraveWalletService: BraveWalletBraveWalletService {
   func setDefaultSolanaWallet(defaultWallet: BraveWallet.DefaultWallet) {
   }
 
-  func discoverAssetsOnAllSupportedChains() {
+  func discoverAssets(onAllSupportedChains bypassRateLimit: Bool) {
   }
```
Comment (nuo-xu) [inline ios/brave-ios/Sources/BraveWallet/Preview Content/MockBraveWalletService.swift:296]:
```suggestion
  func discoverAssetsOnAllSupportedChains(bypassRateLimit: Bool) {
  }
```

File: ios/brave-ios/Sources/BraveWallet/Crypto/Stores/CryptoStore.swift [hunk 4]
```diff
@@ -631,7 +631,7 @@ public class CryptoStore: ObservableObject, WalletObserverStore {
   func prepare(isInitialOpen: Bool = false) {
     Task { @MainActor in
       if isInitialOpen {
-        walletService.discoverAssetsOnAllSupportedChains()
+        walletService.discoverAssets(onAllSupportedChains: true)
```
Comment (nuo-xu) [inline ios/brave-ios/Sources/BraveWallet/Crypto/Stores/CryptoStore.swift:634]:
```suggestion
        walletService.discoverAssetsOnAllSupportedChains(bypassRateLimit: true)
```

**Pull request merged**